File: lending-database/src/main/java/info/armado/ausleihe/database/converter/PrefixConverter.java
/**
 * 
 */
package info.armado.ausleihe.database.converter;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

import info.armado.ausleihe.database.enums.Prefix;

/**
 * @author marc
 *
 */
@Converter(autoApply = true)
public class PrefixConverter implements AttributeConverter<Prefix, String> {

	@Override
	public String convertToDatabaseColumn(Prefix attribute) {
		return attribute.getPrefix();
	}

	@Override
	public Prefix convertToEntityAttribute(String dbData) {
		return Prefix.parse(dbData);
	}

}

File: lending-database/src/main/scala/info/armado/ausleihe/database/converter/PrefixConverter.scala
package info.armado.ausleihe.database.converter

import info.armado.ausleihe.database.dataobjects.Prefix

import javax.persistence.{AttributeConverter, Converter}

/**
  * A converter used to convert [[Prefix]] instances to [[String]]s, which can be persisted inside a database via JPA.
  *
  * @author Marc Arndt
  * @since 24.06.17
  */
@Converter(autoApply = true)
class PrefixConverter extends AttributeConverter[Prefix, String] {
  override def convertToEntityAttribute(databaseValue: String): Prefix = Prefix(databaseValue)

  override def convertToDatabaseColumn(entityValue: Prefix): String = entityValue.prefix
}
